test/smoke/testApps/SpringBootAuto/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/SpringBootAutoTest.java

```diff
@@ -1,13 +1,69 @@
 package com.microsoft.applicationinsights.smoketest;
 
-import org.junit.Test;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import com.google.common.base.Stopwatch;
+import com.microsoft.applicationinsights.internal.schemav2.Data;
+import com.microsoft.applicationinsights.internal.schemav2.Envelope;
+import com.microsoft.applicationinsights.internal.schemav2.RemoteDependencyData;
+import com.microsoft.applicationinsights.internal.schemav2.RequestData;
+import org.junit.*;
+
+import static org.junit.Assert.*;
 
 @UseAgent
 public class SpringBootAutoTest extends AiSmokeTest {
 
     @Test
     @TargetUri("/test")
     public void doMostBasicTest() throws Exception {
-        mockedIngestion.waitForItems("RequestData", 1);
+        List<Envelope> rdList = mockedIngestion.getItemsEnvelopeDataType("RequestData");
+        boolean found = false;
+        for (Envelope envelope : rdList) {
+            RequestData rd = (RequestData) ((Data) envelope.getData()).getBaseData();
+            if (rd.getName().equals("GET /test")) {
+                found = true;
+            }
+        }
+
+        assertTrue(found);
+    }
+
+    @Test
+    @TargetUri("/scheduler")
+    public void fixedRateSchedulerTest() throws Exception {
+        Stopwatch stopwatch = Stopwatch.createStarted();
+        while (mockedIngestion.getCountForType("RequestData") < 20 && stopwatch.elapsed(TimeUnit.SECONDS) < 10) {
+        }
+
+        List<Envelope> rdList = mockedIngestion.getItemsEnvelopeDataType("RequestData");
+        List<Envelope> rddList = mockedIngestion.getItemsEnvelopeDataType("RemoteDependencyData");
+
+        assertTrue(rdList.size() >= 20);
+        for (int i = 0; i < rdList.size(); i++) {
+            Envelope envelope = rdList.get(i);
+            String operationName = envelope.getTags().get("ai.operation.name");
+            RequestData rd = (RequestData) ((Data) envelope.getData()).getBaseData();
+            if (operationName.equals("SpringBootApp.fixedRateScheduler") && rd.getName().equals("SpringBootApp.fixedRateScheduler")) {
```

Comment:
currently if the operationName changes, we won't notice, this conditional will always be false, and test will pass.

does it work to skip the http request and then validate all other (at least 19)?

Reply:
i can remove the check for operationName... checking for name is sufficient.  I just happen to notice that operationname is always the same as requestdata.name.  I can validate others.. but name and target will get changed depended on the type of requests.  here i just check for requests/dependency related to the mapping of '/schedule'